pkg/graph/parser/parser.go

```diff
@@ -75,18 +80,38 @@ func validateSchema(schema *spec.Schema, path string) error {
 	return nil
 }
 
-func getExpectedType(schema *spec.Schema) string {
+func getExpectedType(schema *spec.Schema, resource interface{}) (string, error) {
+	// handle "x-kubernetes-int-or-string"
+	var enabled bool
+	if ext, ok := schema.VendorExtensible.Extensions[xKubernetesIntOrString]; ok {
+		enabled, ok = ext.(bool)
+		if !ok {
+			return "", fmt.Errorf("xKubernetesIntOrString extension is not a boolean")
+		}
+	}
+
+	if enabled {
+		switch resource.(type) {
+		case string:
+			return "string", nil
+		case int:
+			return "integer", nil
+		default:
+			return "", nil
+		}
+	}
+
 	if schema.Type[0] != "" {
-		return schema.Type[0]
+		return schema.Type[0], nil
 	}
 	if schema.AdditionalProperties != nil && schema.AdditionalProperties.Allows {
 		// NOTE(a-hilaly): I don't like the type "any", we might want to change this to "object"
 		// in the future; just haven't really thought about it yet.
 		// Basically "any" means that the field can be of any type, and we have to check
 		// the ExpectedSchema field.
-		return "any"
+		return schemaTypeAny, nil
 	}
-	return ""
+	return "", nil
```

Comment:
Should return an error here? Thinking
```suggestion
	return "", fmt.Errorf("unknown schema type")
```